streampipes-client-go/streampipes/api/data_lake_measure_api.go

```diff
@@ -0,0 +1,121 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"fmt"
+	"streampipes-client-go/streampipes/config"
+	"streampipes-client-go/streampipes/internal/serializer"
+	"streampipes-client-go/streampipes/internal/streampipes_http"
+	"streampipes-client-go/streampipes/model/data_lake"
+)
+
+/*
+DataLakeMeasureApi connects to the DataLakeMeasure endpoint of Streampipes.
+DataLakeMeasureApi supports GET, POST, Delete, and PUT methods for obtaining, deleting, submitting, and updating resources.
+The specific interaction behavior is provided by the method bound to the DataLakeMeasureApi struct.
+Currently, only some GET and Delete methods have been implemented.
+*/
+type DataLakeMeasureApi struct {
+	config      config.StreamPipesClientConnectionConfig
+	httpRequest streampipes_http.HttpRequest
```

Comment:
Why do we need to store the httpRequest in the struct? Seems like it's only used when querying the API.

Reply:
Because dataLakeMeasureApi includes many methods that specifically access the streams endpoint, I placed the httpRequest field inside it

Reply:
Yeah, a new httpReques is allocated each time, but it requires allocating the newly allocated httpReques to httpReques in the datalakemmeasure structure and then calling setResourcePath。First define httpRequest, then call setResourcePath to set the url, this is a sequential relationship

Reply:
ditto https://github.com/apache/streampipes/pull/2437#discussion_r1512941799